Readme.md

```diff
@@ -87,6 +91,36 @@ test-timeout = 300
 test-no-reboot = true
 ```
 
+### Inner workings
+The `bootimage` command first reads the `CARGO_MANIFEST_DIR` environment variable to find out where the `Cargo.toml` of the current project is located.
+Then it parses the `Cargo.toml` file to read bootimage specific configuration data out of it. It then proceeds to build the current cargo project.
+Afterwards it looks if a crate named `bootloader` is defined as dependency. It reads its `Cargo.toml` and looks for more bootimage specific configuration data. These fields must be available:
+```toml
+[features]
+binary = []
```

Comment:
We should add some explanation for this feature:

```suggestion
# Since cargo doesn't support binary-only dependencies, `bootimage` manually turns on
# a feature named `binary`. This way, bootloader crates can use optional dependencies
# to improve their compile times when built as library. 
binary = []
```